java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/utils/Json.java

```diff
@@ -18,89 +18,207 @@
+   *
+   * @param key The key whose value needs to be retrieved.
+   * @return An {@link Optional} Json value if the key exists, else null
+   */
   public Optional<Json> getOptionalJson(final String key) {
     return Optional.ofNullable(document.optJSONObject(key)).map(Json::new);
   }
 
+  /**
+   * Gets the string value associated with a given key.
+   *
+   * @param key The key whose value needs to be retrieved.
+   * @return The value associated with the key, as a string.
+   */
   public String getString(final String key) {
     return document.getString(key);
   }
 
+  /**
+   * Converts the key into a newly-allocated byte array.
+   *
+   * @param key The key whose value needs to be retrieved.
+   * @return The value associated with the key, as a bytes array.
+   */
   public byte[] getBytes(final String key) {
     return Base64.getDecoder().decode(document.getString(key));
   }
 
+  /**
+   * Retrieves the long value associated with a given key and converts it an instance of {@link Instant} using seconds
+   * from the epoch of 1970-01-01T00:00:00Z.
+   *
+   * @param key The key whose value needs to be retrieved.
+   * @return The value time associated with the key.
+   */
   public Instant getInstant(final String key) {
     return Instant.ofEpochSecond(document.getLong(key));
   }
 
+  /**
+   * Gets the boolean value associated with a given key.
+   *
+   * @param key The key whose value needs to be retrieved.
+   * @return An {@code Optional<Boolean>} if it exists, else null.
+   */
   public Optional<Boolean> getOptionalBoolean(final String key) {
     return Optional.ofNullable(document.optBoolean(key));
   }
 
+  /**
+   * Gets the {@link JSONArray} associated with a given key.
+   *
+   * @param key The key whose value needs to be retrieved.
+   * @return The value associated with the key.
+   */
   public JSONArray getJSONArray(final String key) {
     return document.getJSONArray(key);
   }
 
+  /**
+   * Adds a new entry to the {@code Json} object where the value is an {@link java.time.Instant} object.
+   *
+   * @param key The key.
+   * @param instant The value associated with the key.
+   */
   public void put(final String key, final Instant instant) {
     document.put(key, instant.getEpochSecond());
   }
 
+  /**
+   * Adds a new entry to the {@code Json} object where the value is an {@link java.lang.String}.
+   *
+   * @param key The key.
+   * @param string The value associated with the key.
+   */
   public void put(final String key, final String string) {
     document.put(key, string);
   }
 
+  /**
+   * Adds a new entry to the {@code Json} object where the value is a {@link byte} array.
+   *
+   * @param key The key.
+   * @param bytes The value associated with the key.
+   */
   public void put(final String key, final byte[] bytes) {
     document.put(key, Base64.getEncoder().encodeToString(bytes));
   }
 
+  /**
+   * Adds a new entry to the {@code Json} object where the value is a boolean.
```

Comment:
```suggestion
   * Adds a new entry to the {@code Json} object where the value is a {@link bool}.
```

Reply:
bool is not a class so `{@link bool}` would be a dead link.